consensus/service/src/tx_manager.rs

```diff
@@ -510,4 +836,152 @@ mod tests {
+                encrypted_tx: Default::default(),
+                context: context.clone(),
+            };
+
+            tx_manager
+                .well_formed_cache
+                .insert(context.tx_hash().clone(), cache_entry);
+        }
+        assert_eq!(tx_manager.well_formed_cache.len(), tx_hashes.len());
+
+        let aad = vec![];
+        let peer = PeerSession::default();
+        assert!(tx_manager.encrypt_for_peer(&tx_hashes, &aad, &peer).is_ok());
+    }
+
+    #[test_with_logger]
+    // Should return an error if any transaction is not in the cache.
+    fn test_encrypt_for_peer_err_not_in_cache(logger: Logger) {
+        let mock_untrusted = MockUntrustedInterfaces::new();
+        let mock_enclave = MockEnclave::new();
+
+        let tx_manager = TxManagerImpl::new(mock_enclave, mock_untrusted, logger.clone());
+        assert!(tx_manager.well_formed_cache.is_empty());
+
+        let tx_hashes: Vec<_> = (0..10).map(|i| TxHash([i as u8; 32])).collect();
+        let aad = vec![];
+        let peer = PeerSession::default();
+        match tx_manager.encrypt_for_peer(&tx_hashes, &aad, &peer) {
+            Err(TxManagerError::NotInCache(_)) => {} // This is expected.
+            _ => panic!(),
+        }
+    }
+
+    #[test_with_logger]
+    // Should return an error if enclave.txs_for_peer returns an error.
+    fn test_encrypt_for_peer_err_enclave_error(logger: Logger) {
+        let mock_untrusted = MockUntrustedInterfaces::new();
+        let mut mock_enclave = MockEnclave::new();
+
+        // This should be called and should return an error.
+        mock_enclave
+            .expect_txs_for_peer()
+            .times(1)
+            .return_const(Err(EnclaveError::Signature));
+
+        let mut tx_manager = TxManagerImpl::new(mock_enclave, mock_untrusted, logger.clone());
+
+        // Add transactions to the cache.
+        let tx_hashes: Vec<_> = (0..10).map(|i| TxHash([i as u8; 32])).collect();
+        for tx_hash in &tx_hashes {
+            let context = WellFormedTxContext::new(
+                Default::default(),
+                tx_hash.clone(),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+            );
+
+            let cache_entry = CacheEntry {
+                encrypted_tx: Default::default(),
+                context: context.clone(),
+            };
+
+            tx_manager
+                .well_formed_cache
+                .insert(context.tx_hash().clone(), cache_entry);
+        }
+        assert_eq!(tx_manager.well_formed_cache.len(), tx_hashes.len());
+
+        let aad = vec![];
+        let peer = PeerSession::default();
+
+        match tx_manager.encrypt_for_peer(&tx_hashes, &aad, &peer) {
+            Err(TxManagerError::Enclave(EnclaveError::Signature)) => {} // This is expected.
+            _ => panic!(),
+        }
+    }
+
+    #[test_with_logger]
+    #[ignore]
```

Comment:
Was there a particular problem in implementing this test case?

Reply:
No: the get_encrypted_tx method is just really trivial, so I didn't bother with the test. I guess I should just do it.